llarp/config/config.cpp

```diff
@@ -1035,57 +1087,60 @@ namespace llarp
             "The second example binds on the wildcard address using port 9000; the third example",
             "binds on the given IP address using a random high port.",
         },
-        [this, net_ptr, parse_addr_for_link](const std::string& arg) {
-          auto default_port = net::port_t::from_host(DefaultOutboundPort.val);
-          auto addr = parse_addr_for_link(arg, default_port, /*inbound=*/false);
-          if (not addr)
-            addr = net_ptr->WildcardWithPort(default_port);
-          outbound_links.emplace_back(std::move(*addr));
-        });
+        [this, parse_addr_for_link](const std::string& arg) {
+          if (using_new_api)
+            throw std::runtime_error{"USE THE NEW API -- SPECIFY LOCAL ADDRESS UNDER [LISTEN]"};
 
-    conf.add_undeclared_handler(
-        "bind", [this, net_ptr](std::string_view, std::string_view key, std::string_view val) {
-          LogWarn(
-              "using the [bind] section with *=/IP=/INTERFACE= is deprecated; use the inbound= "
-              "and/or outbound= settings instead");
-          std::optional<SockAddr> addr;
-          // special case: wildcard for outbound
-          if (key == "*")
-          {
-            addr = net_ptr->Wildcard();
-            // set port, zero is acceptable here.
-            if (auto port = std::stoi(std::string{val});
-                port < std::numeric_limits<uint16_t>::max())
-            {
-              addr->setPort(port);
-            }
-            else
-              throw std::invalid_argument{fmt::format("invalid port value: '{}'", val)};
-            outbound_links.emplace_back(std::move(*addr));
-            return;
-          }
-          // try as interface name first
-          addr = net_ptr->GetInterfaceAddr(key, AF_INET);
-          if (addr and net_ptr->IsLoopbackAddress(addr->getIP()))
-            throw std::invalid_argument{fmt::format("{} is a loopback interface", key)};
-          // try as ip address next, throws if unable to parse
-          if (not addr)
-          {
-            addr = SockAddr{key, huint16_t{0}};
-            if (net_ptr->IsLoopbackAddress(addr->getIP()))
-              throw std::invalid_argument{fmt::format("{} is a loopback address", key)};
-          }
-          // parse port and set if acceptable non zero value
-          if (auto port = std::stoi(std::string{val});
-              port and port < std::numeric_limits<uint16_t>::max())
-          {
-            addr->setPort(port);
-          }
+          if (auto a = parse_addr_for_link(arg); a and a->is_addressable())
+            addr = a;
           else
-            throw std::invalid_argument{fmt::format("invalid port value: '{}'", val)};
-
-          inbound_listen_addrs.emplace_back(std::move(*addr));
+            addr = oxen::quic::Address{""s, DEFAULT_LISTEN_PORT};
         });
+
+    // DISCUSS: drop this shit?
```

Comment:
I think we have to keep this, unfortunately.  I think there is still quite a bit of use of it in the wild (it wasn't all *that* long ago that we deprecated it).  The `INTERFACE=` option is probably less important to worry about keeping, but `1.2.3.4=5678` and `*=1234` are probably still noticeably used.

Reply:
ugh